src/PrestaShopBundle/Command/ListCommandsAndQueriesCommand.php

```diff
@@ -62,27 +67,130 @@ public function configure()
+            } else {
+                $output->writeln(++$key . '.');
+                $output->writeln('<blue>Class: </blue><info>' . $commandDefinition->getClassName() . '</info>');
+                $output->writeln('<blue>Type: </blue><info>' . $commandDefinition->getCommandType() . '</info>');
+                $output->writeln('<blue>API: </blue><info>' . $cqrsEndpointURI . '</info>');
+                $output->writeln('<comment>' . $commandDefinition->getDescription() . '</comment>');
+                $output->writeln('');
+            }
         }
 
         return 0;
     }
+
+    private function handleOptions(InputInterface $input): void
+    {
+        if ($input->getOption('domain') !== []) {
+            $this->filterCQRS($input->getOption('domain'));
+        }
+
+        if ($input->getOption('format') === 'simple') {
+            $this->isFormatSimple = true;
+        }
+    }
+
+    /**
+     * @param string[] $filters
+     */
+    private function filterCQRS(array $filters): void
+    {
+        $this->commandAndQueries = array_filter($this->commandAndQueries, function (string $currentCQRS) use ($filters) {
+            foreach ($filters as $filter) {
+                // We append a backslash behind the filter to find only exact matches
+                if (str_contains($currentCQRS, $filter . '\\') !== false) {
+                    return true;
+                }
+            }
+
+            return false;
+        });
+    }
+
+    /**
+     * This method rebuild the resources from the api platform attributes.
+     *
+     * @return ResourceMetadataCollection[]
+     */
+    private function getResourceList(): array
+    {
+        $resourceMetadataCollection = [];
+        foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
+            $resourceMetadataCollection[] = $this->resourceMetadataFactory->create($resourceClass);
+        }
+
+        return $resourceMetadataCollection;
+    }
+
+    /**
+     * This method takes the filtered list of routes, and checks it against the list of CQRS.
+     * The CQRS that have a route with the correct method are implemented, so we return the URI of the endpoint.
+     */
+    private function getCQRSEndpointURI(CommandDefinition $commandDefinition): string
+    {
+        $domain = explode('\\', $commandDefinition->getClassName())[4];
+
+        foreach ($this->apiResourcesList as $apiResources) {
+            foreach ($apiResources as $resource) {
+                if ($resource->getShortName() !== $domain) {
+                    return '';
+                }
+
+                $apiResourceOperations = $resource->getOperations();
+                foreach ($apiResourceOperations as $operation) {
+                    if ($this->doesMethodsMatchType($operation->getMethod(), $commandDefinition->getCommandType())) {
+                        return $operation->getUriTemplate();
```

Comment:
What happens if `doesMethodsMatchType()` returns false? We simply skip it?

It might be useful then to add the number of skipped/invalid records to the output of the Command, this might be a good information for the user.

Imagine you run the Command and you get a empty output but it also tells you it skipped 300 records 😁 then you know something wrong happened

Reply:
Currently there is no easy way to detect if the current operation is a command or a query. This method is just here to diferenciate the two of them so you won't return the URI of the command when handling the query and vice versa.

The list of CQRS in the input is gathered from the tagged services. Unless someone did something very wrong with the tags or the compiler pass it will never have 300 skipped records only one every time ( in case we currently handle the query the comand is skipped ... )